SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -15,21 +17,52 @@ import kotlinx.serialization.json.Json
 import kotlinx.serialization.json.JsonConfiguration
 import kotlin.coroutines.CoroutineContext
 
+import com.jetbrains.handson.mpp.mobile.api.*
+import kotlinx.serialization.json.JsonBuilder
+
+@ImplicitReflectionSerializer
 class ApplicationPresenter : ApplicationContract.Presenter() {
 
     private val dispatchers = AppDispatchersImpl()
     private var view: ApplicationContract.View? = null
     private val job: Job = SupervisorJob()
-    private val codeMap = mapOf<String, String>(
+    private val codeMap =
+        mutableMapOf<String, String>(
         "Harrow and Wealdstone" to "HRW",
         "Canley" to "CNL",
         "London Euston" to "EUS",
         "Coventry" to "COV",
         "Birmingham New Street" to "BHM"
     )
+
+    init {
+        launch {
+            codeMap.putAll(
+                client.getStations().stations.map {
+                    it.name to (it.nlc ?: it.crs ?: throw Error())
+                }.toMap()
+            )
+            val stations = client.getStations().stations
+            for ( i in stations) {
+                println(i)
+            }
+        }
+    }
```

Comment:
If this works, do we need to start with hard-coded map of stations above?

Reply:
It doesn't work yet, since I need to figure out how to update the spinner's list (we're going to switch to using an autocomplete text box) kind of asynchronously, so I left it for now

Reply:
Alright, happy for hard coded stations to be left in for now